src/uct/ib/base/ib_iface.c

```diff
@@ -923,8 +923,26 @@ static ucs_status_t uct_ib_iface_init_lmc(uct_ib_iface_t *iface,
     return UCS_OK;
 }
 
+static void
+uct_ib_iface_adjust_qp_attr(uct_ib_iface_t *iface, uct_ib_qp_attr_t *qp_attr)
+{
+    uct_ib_device_t *dev    = uct_ib_iface_device(iface);
+    struct ibv_qp_cap *caps = &qp_attr->ibv.cap;
+
+    uct_ib_iface_check_cap_limit(iface, IBV_DEV_ATTR(dev, max_qp_wr),
+                                 "max_send_wr", &caps->max_send_wr);
+    uct_ib_iface_check_cap_limit(iface, IBV_DEV_ATTR(dev, max_qp_wr),
+                                 "max_recv_wr", &caps->max_recv_wr);
+    uct_ib_iface_check_cap_limit(iface, IBV_DEV_ATTR(dev, max_sge),
+                                 "max_send_sge", &caps->max_send_sge);
+    uct_ib_iface_check_cap_limit(iface, IBV_DEV_ATTR(dev, max_sge),
+                                 "max_recv_sge", &caps->max_recv_sge);
+}
+
 void uct_ib_iface_fill_attr(uct_ib_iface_t *iface, uct_ib_qp_attr_t *attr)
 {
+    memset(&attr->ibv, 0, sizeof(attr->ibv));
```

Comment:
Otherwise it's an uninitialized struct, which can be either `ibv_qp_init_attr`, `ibv_qp_init_attr_ex` or `ibv_exp_qp_init_attr`, and it fails without this memset on one of the NICs I tested (not a Mellanox NIC).

Reply:
looks like the callers of uct_ib_iface_fill_attr() already initialize the attr struct, in which case is it not initialized?

Reply:
Not all fields are initialized (e.g. `qp_context` for `ibv_qp_init_attr` and `comp_mask` in `ibv_qp_init_attr_ex`), and I'm afraid I no longer have access to the NIC in question... I don't mind (asking Shuki to) remove this memset (or set just these two fields I mentioned) in a follow-up patch, but I can't tell if this leaves UCX working on said NICs.

Reply:
For example here it's set to 0:
https://github.com/openucx/ucx/blob/0ee08a4d686c6fd3e07983992013e5fce4a21c00/src/uct/ib/rc/verbs/rc_verbs_ep.c#L598

Question if this PR added new flow where attr is not initialized (I don't see such), or there already was some flow where it wasn't initialized (didn't find such as well)
@shuki-zanyovka can you pls check?